src/ng-select/ng-option-highlight.directive.ts

```diff
@@ -0,0 +1,42 @@
+import {Directive, ElementRef, Input, OnChanges, Renderer2} from '@angular/core';
+import * as searchHelper from './search-helper';
+
+@Directive({
+    selector: '[ngOptionHighlight]'
+})
+export class NgOptionHighlightDirective implements OnChanges {
```

Comment:
Looks nice, only add some unit tests for this directive and I can merge it. You can create new test file called `ng-option-highlight.directive.spec.ts`.

Reply:
No problem I will add some unit tests tonight